### PR TITLE
refac(impression-area): Improved ImpressionArea and useImpressionRef

### PR DESCRIPTION
**Pull request by ssi02014**
## Overview

I worked on refactoring `ImpressionArea` and `useImpressionRef`. All tests worked fine. 

I left a description of the task as a comment..!! 
I'd appreciate it if you could review it. 🙏

## PR Checklist

- [x] I read and included theses actions below

1. I have read the [Contributing Guide](https://github.com/toss/slash/blob/main/.github/CONTRIBUTING.md)
2. I have written documents and tests, if needed.

### CODE REVIEW

File: packages/react/impression-area/src/ImpressionArea.tsx [hunk 1]
```diff
@@ -1,13 +1,11 @@
 /** @tossdocs-ignore */
 import { useCombinedRefs } from '@toss/react';
-import { forwardRef, Ref } from 'react';
+import { forwardRef, PropsWithChildren } from 'react';
 import { useImpressionRef } from './useImpressionRef';
 
-export interface ImpressionAreaProps {
+export interface ImpressionAreaProps extends Omit<IntersectionObserverInit, 'threshold'> {
```
Comment (ssi02014) [inline packages/react/impression-area/src/ImpressionArea.tsx:6]: Improved `ImpressionAreaProps` by extending `IntersectionObserverInit`.

```ts
interface IntersectionObserverInit {
    root?: Element | Document | null;
    rootMargin?: string;
    threshold?: number | number[];
}
```

File: packages/react/impression-area/src/ImpressionArea.tsx [hunk 2]
```diff
@@ -21,14 +19,13 @@ export interface ImpressionAreaProps {
...
   className?: string;
-  children?: React.ReactNode;
   style?: React.CSSProperties;
   role?: string;
   'aria-label'?: string;
   'aria-hidden'?: boolean | 'false' | 'true';
 }
 
-export const ImpressionArea = forwardRef(
+export const ImpressionArea = forwardRef<HTMLDivElement, PropsWithChildren<ImpressionAreaProps>>(
```
Comment (ssi02014) [inline packages/react/impression-area/src/ImpressionArea.tsx:28]: We can utilize the generics of forwardRef.

```ts
function forwardRef<T, P = {}>(render: ForwardRefRenderFunction<T, P>): ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>>;
```

File: packages/react/impression-area/src/useImpressionRef.tsx [hunk 3]
```diff
@@ -11,17 +11,18 @@ type Options = Pick<
...
+  onImpressionEnd: _onImpressionEnd = noop,
   timeThreshold = 0,
   root,
   rootMargin,
   areaThreshold: intersectThreshold = 0,
 }: Options) {
-  const onImpressionStart = usePreservedCallback(_onImpressionStart ?? noop);
-  const onImpressionEnd = usePreservedCallback(_onImpressionEnd ?? noop);
+  const onImpressionStart = usePreservedCallback(_onImpressionStart);
+  const onImpressionEnd = usePreservedCallback(_onImpressionEnd);
```
Comment (ssi02014) [inline packages/react/impression-area/src/useImpressionRef.tsx:22]: What about specifying a `default parameter` in the form of something like `areaThreshold`?

File: packages/react/impression-area/src/useImpressionRef.tsx [hunk 4]
```diff
@@ -49,33 +50,17 @@ export function useImpressionRef<Element extends HTMLElement>({
...
 
-  useDocumentVisibilityChange(documentVisible => {
+  useVisibilityEvent(visibilityState => {
     if (!isIntersectingRef.current) {
       return;
     }
 
-    setDebouncedIsImpressed(documentVisible);
+    setDebouncedIsImpressed(visibilityState === 'visible');
   });
```
Comment (ssi02014) [inline packages/react/impression-area/src/useImpressionRef.tsx:59]: We can reuse `useVisibilityEvent` which is already defined in `@toss/react`.

File: packages/react/impression-area/src/useImpressionRef.tsx [hunk 5]
```diff
@@ -111,10 +96,13 @@ function useIntersectionObserver<Element extends HTMLElement>(
...
 
-function useSetDebouncedBooleanValue(options: { onValueChange: (newValue: boolean) => void; timeThreshold: number }) {
+function useSetDebouncedBooleanValue(options: {
+  onValueChange: (newValue: boolean) => void;
+  timeThreshold: Options['timeThreshold'];
+}) {
   const { onValueChange, timeThreshold } = options;
   const handleValueChange = usePreservedCallback(onValueChange);
-  const ref = useRef({ value: false, cancelPrevDebounce: () => { } });
+  const ref = useRef({ value: false, cancelPrevDebounce: noop });
```
Comment (ssi02014) [inline packages/react/impression-area/src/useImpressionRef.tsx:105]: I reused a `noop` function that was already defined.

**Review by okinawaa — APPROVED** [review]
thanks! 
LGTM 🏋️

**Comment by ssi02014** [comment]
@okinawaa thanks!!!👏👏🏋🏋

**Pull request merged**